contracts/PetAdoption.sol

```diff
@@ -1,42 +1,103 @@
 // SPDX-License-Identifier: UNLICENSED
-pragma solidity ^0.8.9;
-// import "hardhat/console.sol";
+pragma solidity ^0.8.27;
 
 contract PetAdoption{
-    address public owner;
-    uint public petIndex = 0;
-    uint[] public allAdoptedPets;
+    // Struct to store pet details
+    struct Pet {
+        uint256 id;
+        string category;
+        string breed;
+    }
+    address private immutable i_owner;
+    // Number of pets added, will also be used as pet id and index
+    uint256 private petsNumber;
 
-    mapping(uint => address) public petToOwner;
-    mapping(address => uint[]) public ownerToPets;
+    Pet[] private allPets;
+    mapping(uint256 => address) private petIdToOwner;
+    mapping(address => uint256[]) private ownerToPetIds;
 
-    constructor(uint initialPetIndex){
-        owner = msg.sender;
-        petIndex = initialPetIndex; 
+    constructor(){
+        i_owner = msg.sender;
```

Comment:
why you are changing the name to `i_owner`?

Reply:
Just a standard practice. It shows that owner is immutable ( hence "i_").